src/tribler-core/run_tribler_headless.py

```diff
@@ -82,6 +93,7 @@ def log_incoming_remote_search(self, sock_addr, keywords):
         if options.restapi > 0:
             config.set_http_api_enabled(True)
             config.set_http_api_port(options.restapi)
+            config.write()  # only write config when rest api is enabled
```

Comment:
From the user's standpoint, it is not obvious that the config should be written if REST API is enabled. Overall, we want the config to be read-only and command-line options to override the config options.

Please, remove this line.

Reply:
how to get the rest apikey then? there is no option to push custom apikey to rest api key from run time arguments, afaik it is generated by the daemon and written stored in the config object.

The purpose is to receive the apikey so the users can connect.

I am referencing this comment

> # Immediately write the API key so other applications can use it

https://github.com/Tribler/tribler/blob/09fbd720a6c251ffa76bfb2d2560b60a6920994d/src/run_tribler.py#L82

Reply:
@egbertbouman , could you please comment on the API key usage?

Reply:
The comment you're referring to is just a really ugly hack so that developers have easy access to the API key. You don't need to do this in `run_tribler_headless.py`.

By default the core does not use an API key. So, if you want to use one you have to set is yourself (using `config.set_http_api_key(apikey)`).

Reply:
Ok, i understood the story, since this part is on production i thought it was ok to write api key, but on broader scale this is definetely a security leak to write pass to filesystem like that.

Reply:
I wasn't trying to say that you can't store API key in the config file, just that you don't need to write the config file here.

Reply:
Yep, got it, config obj is on ram unless u write it. But headless does not have argument to pass it. Nevermind i think everyone are in the same page.